### DISCUSSION TITLE
Weekly notes name template

### DISCUSSION THREAD

**Discussion by anpin** [zk-org/zk#315] [Help] [answered]
Hi, first of all, thanks a lot  for this amazing productivity booster I'm truly enjoying using it. 
I want to dive into current week journal with one liner command. How should one approach `filename` template for a weekly journal, so it would create some stable human readable name such as first day of the week or maybe use an alias to open already created file for the current week and if non exists create a new one?

**Comment by mickael-menu** [comment, marked as answer]
It depends what you want exactly, but you can use a third-party shell command to generate the filename. For example using the `date` terminal command:

```toml
filename = "{{sh 'date +%A'}}"
```

> use an alias to open already created file for the current week and if non exists create a new one?

This is already the default behavior of `zk new` with the `--no-input` flag. If the filename already exists, then it is opened instead of overwriting it.

**Reply by anpin** [reply]
thank you so much! I haven't realized I can use shell commands for file names.

**Comment by tjex** [comment]
for future visitors, this idea is written here in more detail:
~https://github.com/mickael-menu/zk/blob/main/docs/daily-journal.md~ (see below)

**Reply by zashishz** [reply]
This above URL is changed to https://github.com/zk-org/zk/tree/main/docs/tips
Just in case any one like me getting started with zk.
Thanks for the amazing work @mickael-menu and team.

**Reply by tjex** [reply]
Or here rendered in pretty html: https://zk-org.github.io/zk/tips/
